core/src/main/scala/zio/logging/ContextMap.scala

```diff
@@ -2,7 +2,21 @@ package zio.logging
 
 import zio._
 
-final class ContextMap private (private val map: FiberRef[ContextMap.CMap]) {
+trait LoggingContext {
+  def loggingContext: LoggingContext.Service[Any]
+}
+
+object LoggingContext {
+  trait Service[-R] {
+    def get[V](key: ContextKey[V]): ZIO[R, Nothing, V]
+    def add[V](key: ContextKey[V], value: V): ZIO[R, Nothing, Unit]
+    def set[V](key: ContextKey[V], value: V): ZIO[R, Nothing, Unit]
```

Comment:
The difference between add and set is confusing. I think it should only be `add` and the particular `ContextKey` can decide whether to "append" or "replace".

Reply:
I will remove ```add```